app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -4144,9 +4144,15 @@ private void useVideoSource(final boolean videoEnabled) {
 
         final int videoRenderIndex = getVideoRendererIndex();
 
-        // We can safely assume that currentMetadata is not null (otherwise this method isn't
-        // called) so we can use the requireNonNull method of the Objects class.
-        final StreamInfo info = Objects.requireNonNull(currentMetadata).getMetadata();
+        // The current metadata may be null sometimes so we will be not able to execute the
+        // block above. Reload the play queue manager in this case.
```

Comment:
I think you mean below or?
Could you also note that this happens e.g. on unstable network connections.
You may also note that you reload the play queue manager because it's the default behavior without the optimization.

Reply:
> I think you mean below or?

Yes, I mismatched the words in my mind 😅. Fixed with https://github.com/TeamNewPipe/NewPipe/pull/7349/commits/6341c14c870885a48512479a00e65dac78d0ef7c.